pkg/gardenlet/controller/managedseed/managedseed_actuator.go

```diff
@@ -345,13 +346,18 @@ func (a *actuator) deployGardenlet(
 	gardenletConfig *configv1alpha1.GardenletConfiguration,
 	shoot *gardencorev1beta1.Shoot,
 ) error {
+	seed, err := a.getSeed(ctx, managedSeed)
+	if err != nil {
+		return err
+	}
+
```

Comment:
For consistency, I would put this code in `Reconcile` and pass the seed to this function as a parameter, as it's done in `deleteGardenlet`. (to maintain the symmetry between `deployGardenlet` and `deleteGardenlet`).